python-mini-projects/projects/Scrape_Hacker_News/main.py

```diff
@@ -0,0 +1,99 @@
+import requests
+import os
+from bs4 import BeautifulSoup, SoupStrainer
+# Makes Output Directory if it does not exist
+if not os.path.exists(os.path.join(os.getcwd(), 'HackerNews')):
+    os.makedirs(os.path.join(os.getcwd(), 'HackerNews'))
+'''
+@params page_no: The page number of HackerNews to fetch.
+Adding only page number in order to add multiprocess support in future.
+@params verbose: Adds verbose output to screen instead
+of running the program silently.
+'''
+
+
+def fetch(page_no, verbose=False):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 57 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.